src/simulators/statevector/batched_matrix.hpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2018, 2019, 2020, 2021.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+
+#ifndef _qv_batched_matrix_hpp_
+#define _qv_batched_matrix_hpp_
+
+#include "framework/operations.hpp"
+
+namespace AER {
+namespace QV {
+
+class alignas(128) batched_matrix_params 
+{
+public:
+  std::complex<double> matrix2x2_[4];
```

Comment:
Most of operations can be calculated by 2x2 so this is used to avoid storing matrix to GPU. Larger matrices (e.g. 4x4) are stored in the array and `offset_matrix_` points the top of stored matrix.